edn/src/matcher.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use std::collections::HashMap;
+use regex::Regex;
+use itertools::diff_with;
+
+use types::Value;
+
+///
+pub fn match_values<T>(pattern: T, input: T) -> bool where T: AsRef<Value> {
+    use Value::*;
```

Comment:
Do we have a type that is `AsRef<Value>` other than `Value`?  It can't be `ValueAndSpan` because the inner children are wrong...